src/attributes/attribute_factory.rs

```diff
@@ -0,0 +1,39 @@
+use crate::constants::constant_info::ConstantInfo;
+use crate::constants::utf8_info::Utf8Info;
+use crate::attributes::attribute_info::{AttributeInfo};
+use crate::util::to_u16;
+use crate::attributes::code_attribute::CodeAttribute;
+use crate::attributes::constant_value_attribute::ConstantValueAttribute;
+use crate::attributes::deprecated_attribute::DeprecatedAttribute;
+use crate::attributes::signature_attribute::SignatureAttribute;
+use crate::attributes::exception_attribute::ExceptionAttribute;
+use std::collections::VecDeque;
+use crate::vecdeque;
+
+const CONSTANT_VALUE: &str = "ConstantValue";
+const CODE: &str = "Code";
+const DEPRECATED: &str = "Deprecated";
+const SIGNATURE: &str = "Signature";
+const EXCEPTION: &str = "Exception";
+
+pub fn get_attribute(mut data: &mut VecDeque<u8>, constant_pool: &[Box<dyn ConstantInfo>]) -> Box<dyn AttributeInfo>
```

Comment:
@jmanno1 So i learned with the Vec<u8> implementation for the byte stream that the vector wasnt actually being consumed.
I changed the implementation to a VecDeque<u8> and mutate the data structure, I like this implementation since it requires absolutely no lifetimes, one copy of the data, and most importantly consumes the byte stream one byte at a time. 

Im curious to hear your thoughts, but boy oh boy that was an expensive mistake, spent like 2 hours copy and pasting 'mut' everywhere

Reply:
@cjhillbrand I like the idea of the DeQue as well. 

I wasn't sure if it was your intention to mutate the data or not with the Vec implementation, I agree this one is a good one though.